os/pm/pm_idle.c

```diff
@@ -84,7 +94,35 @@ void pm_idle(void)
 		/* Decide, which power saving level can be obtained */
 		newstate = pm_checkstate();
 		/* Perform state-dependent logic here */
-		pmvdbg("newstate= %d\n", newstate);
+		/* For SMP case, we need to check secondary core status
+	     * If secondary core status is not in idle thread, abort
+	     * the sleep and check again on next cycle
+		 */
+#ifdef CONFIG_SMP
+		if (newstate == PM_SLEEP) {
```

Comment:
Please move this section below `pm_metrics` code

Reply:
sure